test/e2e/case1_framework_deployment_test.go

```diff
@@ -143,6 +144,58 @@ var _ = Describe("Test framework deployment", func() {
 		}
 	})
 
+	It("should create a framework deployment with node selector on the managed cluster", func() {
+		By("Creating the AddOnDeploymentConfig")
+		Kubectl("apply", "-f", addOnDeplomentConfigCR)
+		By("Creating the governance-policy-framework ClusterManagementAddOn to use the AddOnDeploymentConfig")
+		Kubectl("apply", "-f", case1ClusterManagementAddOnCR)
+
+		for i, cluster := range managedClusterList {
+			logPrefix := cluster.clusterType + " " + cluster.clusterName + ": "
+			By(logPrefix + "deploying the default framework managedclusteraddon")
+			if cluster.clusterType == "hub" {
+				Kubectl("apply", "-n", cluster.clusterName, "-f", case1hubAnnotationMCAOCR)
+			} else {
+				Kubectl("apply", "-n", cluster.clusterName, "-f", case1ManagedClusterAddOnCR)
+			}
+
+			deploy := GetWithTimeout(
+				cluster.clusterClient, gvrDeployment, case1DeploymentName, addonNamespace, true, 60,
+			)
+			Expect(deploy).NotTo(BeNil())
+
+			checkContainersAndAvailability(cluster, i)
+
+			By(logPrefix + "verifying the nodeSelector")
+			nodeSelector, _, _ := unstructured.NestedStringMap(
+				deploy.Object, "spec", "template", "spec", "nodeSelector",
+			)
+			Expect(nodeSelector).To(Equal(map[string]string{"kubernetes.io/os": "linux"}))
+
+			By(logPrefix + "verifying the tolerations")
+			tolerations, _, _ := unstructured.NestedSlice(deploy.Object, "spec", "template", "spec", "tolerations")
+			Expect(tolerations).To(HaveLen(1))
+			expected := map[string]interface{}{
+				"effect":   "NoSchedule",
+				"key":      "node.kubernetes.io/unreachable",
+				"operator": "Exists",
+			}
+			Expect(tolerations[0]).To(Equal(expected))
```

Comment:
Maybe I'm being paranoid, but these seem like very safe default values, is it possible to change them to be more obviously tests, to double-verify that they are only being set because of the configuration? Alternatively, I think we could have another test that just shows that these are *not* set like this without the configuration (or maybe another test already does that and I missed it).